### PR TITLE
Adding emissary e2e tests

### PR DESCRIPTION
**Pull request by jonahjon**
Signed-off-by: jonahjon <jonahjones094@gmail.com>

*Issue #, if available:*

*Description of changes:*

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/e2e/emissary_test.go [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+
+const (
+	emissaryPackageName   = "emissary"
+	emissaryPackagePrefix = "test"
+)
+
+func runCuratedPackageEmissaryInstall(test *framework.ClusterE2ETest) {
+	test.SetPackageBundleActive()
+	packageFile := test.BuildPackageConfigFile(emissaryPackageName, emissaryPackagePrefix, EksaPackagesNamespace)
+	test.KubectlClient.WaitForJobCompleted(context.TODO(), kubeconfig.FromClusterName(test.ClusterName), "1m", "complete", "eksa-auth-refresher", EksaPackagesNamespace)
```
Comment (junshun) [inline test/e2e/emissary_test.go:25]: Remove this as latest CLI build no longer run a one time job.

File: test/e2e/emissary_test.go [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+	test.DeleteCluster()
+	test.PowerOffHardware()
+	test.ValidateHardwareDecommissioned()
+}
+
+func runCuratedPackageEmissaryInstallSimpleFlow(test *framework.ClusterE2ETest) {
+	test.WithCluster(runCuratedPackageEmissaryInstall)
+}
+
+func TestCPackagesEmissaryDockerUbuntuKubernetes120SimpleFlow(t *testing.T) {
```
Comment (junshun) [inline test/e2e/emissary_test.go:64]: Remove this as packages don't support 120 so this test wont pass.

File: test/e2e/emissary_test.go [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+		framework.NewDocker(t),
+		framework.WithClusterFiller(api.WithKubernetesVersion(v1alpha1.Kube122)),
+		framework.WithPackageConfig(t, packageBundleURI(v1alpha1.Kube122),
+			EksaPackageControllerHelmChartName, EksaPackageControllerHelmURI,
+			EksaPackageControllerHelmVersion, EksaPackageControllerHelmValues),
+	)
+	runCuratedPackageEmissaryInstallSimpleFlow(test)
+}
+
+func TestCPackagesEmissaryVSphereKubernetes120SimpleFlow(t *testing.T) {
```
Comment (junshun) [inline test/e2e/emissary_test.go:100]: Same

File: test/e2e/emissary_test.go [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+		framework.NewDocker(t),
+		framework.WithClusterFiller(api.WithKubernetesVersion(v1alpha1.Kube120)),
+		framework.WithPackageConfig(t, packageBundleURI(v1alpha1.Kube120),
+			EksaPackageControllerHelmChartName, EksaPackageControllerHelmURI,
+			EksaPackageControllerHelmVersion, EksaPackageControllerHelmValues),
+	)
+	runCuratedPackageEmissaryInstallSimpleFlow(test)
+}
+
+func TestCPackagesEmissaryDockerUbuntuKubernetes121SimpleFlow(t *testing.T) {
```
Comment (junshun) [inline test/e2e/emissary_test.go:76]: We should probably have separate discussion on if we want to run packages e2e on all kube versions. I had though that we were only going to do that for hello as a smoke test.

**Review by junshun — CHANGES_REQUESTED** [review]

File: test/e2e/emissary_test.go (hunk 4 above)
Comment (jonahjon) [reply test/e2e/emissary_test.go]: Fixed

File: test/e2e/emissary_test.go (hunk 3 above)
Comment (jonahjon) [reply test/e2e/emissary_test.go]: Fixed

File: test/e2e/emissary_test.go (hunk 2 above)
Comment (jonahjon) [reply test/e2e/emissary_test.go]: Fixed

File: test/e2e/emissary_test.go (hunk 1 above)
Comment (jonahjon) [reply test/e2e/emissary_test.go]: Fixed

**Review by junshun — APPROVED** [review]

**Pull request merged**